app/src/main/java/io/homeassistant/companion/android/sensors/NetworkSensorManager.kt

```diff
@@ -375,4 +407,34 @@ class NetworkSensorManager : SensorManager {
                 (ip shr 16 and 0xFF) + "." +
                 (ip shr 24 and 0xFF)
     }
+
+    private fun updatePublicIpSensor(context: Context) {
+        var ip = "unknown"
```

Comment:
You are missing the required permission call here I think so it may be forcing the sensor to update and become enabled.

See an example: https://github.com/home-assistant/android/blob/master/app/src/main/java/io/homeassistant/companion/android/sensors/NetworkSensorManager.kt#L91

Also you may want to add a permission check like we are doing here: https://github.com/home-assistant/android/blob/master/app/src/main/java/io/homeassistant/companion/android/sensors/NetworkSensorManager.kt#L99

Reply:
Not sure if the last permission check is required but I am pretty sure the first one is.

Reply:
Yep, first check is the reason why it become enabled. 
Permission check is not required.